DuckDuckGo/MainViewController.swift

```diff
@@ -131,6 +141,12 @@ class MainViewController: UIViewController {
         animateForKeyboard(userInfo: userInfo, y: view.frame.height)
     }
     
+    @objc func onStorageCacheChange() {
+        DispatchQueue.main.async {
+            self.storageCache = AppDependencyProvider.shared.storageCache.current
```

Comment:
Yes, this confused me as well ;p

From what I understand, trigger to do so does not come from MainViewController, but from the TabController, as it is the one providing the siteRating. And since TabController is also listening for this notification, it will be reloaded, thus propagating changes to SiteRating if needed.